src/types/script.zig

```diff
@@ -0,0 +1,56 @@
+const std = @import("std");
+const CompactSizeUint = @import("bitcoin-primitives").types.CompatSizeUint;
+const readBytesExact = @import("../util/mem/read.zig").readBytesExact;
+
+bytes: []u8,
+allocator: std.mem.Allocator,
+
+const Self = @This();
+
+/// Initialize a new script
+pub fn init(allocator: std.mem.Allocator) !Self {
+    return Self{
+        .bytes = try allocator.alloc(u8, 0),
+        .allocator = allocator,
+    };
+}
+
+/// Deinitialize the script
+pub fn deinit(self: *Self) void {
+    self.allocator.free(self.bytes);
+}
+
+/// Add data to the script
+pub fn push(self: *Self, data: []const u8) !void {
+    const new_len = self.bytes.len + data.len;
+    self.bytes = try self.allocator.realloc(self.bytes, new_len);
```

Comment:
sp use an arrayList to have a growable slice

Reply:
That's what I said in the previous comments haha

Reply:
Should I make it back into ArrayList ?

Reply:
I think so. Even if #113 will provide better usability